File: src/main/java/de/florianschmitt/system/configuration/ProductiveDataSourceConfiguration.java
package de.florianschmitt.system.configuration;

import com.zaxxer.hikari.HikariDataSource;
import de.florianschmitt.system.util.ProductiveProfile;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import javax.sql.DataSource;
import java.net.URI;
import java.net.URISyntaxException;

@Configuration
@ProductiveProfile
public class ProductiveDataSourceConfiguration {

    @Bean
    public DataSource dataSource() throws URISyntaxException {
        URI dbUri = new URI(System.getenv("DATABASE_URL"));

        String username = dbUri.getUserInfo().split(":")[0];
        String password = dbUri.getUserInfo().split(":")[1];
        String dbUrl = "jdbc:postgresql://" + dbUri.getHost() + ':' + dbUri.getPort() + dbUri.getPath();

        HikariDataSource basicDataSource = new HikariDataSource();
        basicDataSource.setJdbcUrl(dbUrl);
        basicDataSource.setUsername(username);
        basicDataSource.setPassword(password);

        return basicDataSource;
    }
}

File: src/main/kotlin/de/florianschmitt/system/configuration/ProductiveDataSourceConfiguration.kt
package de.florianschmitt.system.configuration

import com.zaxxer.hikari.HikariDataSource
import de.florianschmitt.system.util.ProductiveProfile
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.net.URI
import java.net.URISyntaxException
import javax.sql.DataSource

@Configuration
@ProductiveProfile
class ProductiveDataSourceConfiguration {

    @Bean
    @Throws(URISyntaxException::class)
    fun dataSource(): DataSource {
        val dbUri = URI(System.getenv("DATABASE_URL"))

        val username = dbUri.userInfo.split(":")[0]
        val password = dbUri.userInfo.split(":")[1]

        val dataSource = HikariDataSource()
        dataSource.jdbcUrl = "jdbc:postgresql://${dbUri.host}:${dbUri.port}${dbUri.path}"
        dataSource.username = username
        dataSource.password = password

        return dataSource
    }
}
